src/pages/api/auth/[...nextauth].js

```diff
@@ -1,10 +1,100 @@
+import { MongoDBAdapter } from "@next-auth/mongodb-adapter";
 import NextAuth from "next-auth";
+import dbConnect, { DB_INFO } from "../../../../server/db/dbConnect";
+import GoogleProvider from "next-auth/providers/google";
 
+let cachedPromise;
+
+import { MongoClient, ObjectId } from "mongodb";
+import User from "../../../../server/db/models/User";
+import { consts } from "@/utils/consts";
+
+const client = new MongoClient(DB_INFO.DB_CONNECTION_STRING);
+
+const retrievePromise = () => {
+  if (cachedPromise) {
+    return cachedPromise;
+  } else {
+    cachedPromise = client.connect();
+    return cachedPromise;
+  }
+};
+
+/**
+ * @type {import("next-auth").NextAuthOptions}
+ */
 export const authOptions = {
-  // Configure one or more authentication providers
+  adapter: MongoDBAdapter(retrievePromise()),
+
+  // Use JWTs for everything session-related
+  session: {
+    strategy: "jwt",
+  },
+  secret: process.env.NEXTAUTH_SECRET,
   providers: [
-    // ...add more providers here
+    GoogleProvider({
+      clientId: process.env.GOOGLE_CLIENT_ID,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+    }),
   ],
+  pages: {
+    signIn: "/login",
+    newUser: "/dogs",
+    error: "/login",
+  },
+  events: {
+    createUser: async (message) => {
+      await dbConnect();
+
+      const _id = new ObjectId(message.user.id);
+
+      // TODO: Find a way of avoiding unnecessary deletion calls
+      await User.deleteOne({ _id });
```

Comment:
this is not too much of a problem tbh. the operation only happens when new users are created, so not very often

Reply:
sure